core/types/block.go

```diff
@@ -294,7 +296,14 @@ func NewBlock(header *Header, body *Body, receipts []*Receipt, hasher TrieHasher
 		}
 	}
 
-	if withdrawals == nil {
+	if config.IsIsthmus(header.Time) {
+		if withdrawals == nil || len(withdrawals) > 0 {
+			log.Crit("expected non-nil empty withdrawals operation list in Isthmus, but got: %v", body.Withdrawals)
+		}
```

Comment:
A `panic` is better than a `log.Crit`, since `log.Crit` has program-exit behavior, which is bad with testing and unexpected user input.